presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -499,6 +500,19 @@ public HiveConfig setRcfileWriterValidate(boolean rcfileWriterValidate)
         return this;
     }
 
+    public boolean isRcfilePageSourceValidateFileSize()
+    {
+        return rcfilePageSourceValidateFileSize;
+    }
+
+    @Config("hive.rcfile.page-source-validate-file-size")
+    @ConfigDescription("Recheck RCFile file size during page source creation to handle client side encrypted files reporting larger than actual sizes")
```

Comment:
This is a bit long and detailed for description which is intended to be a synopsis. How about

> Recheck RCFile file size on open to handle client side encrypted files

Also, would server side encrypted files have the same issue? (I'm not sure if those are completely transparent on the S3 side)

Reply:
Reworded to just "encrypted files.